addons/sourcemod/scripting/freak_fortress_2.sp

```diff
@@ -6555,274 +6556,151 @@ stock GetBossIndex(client)
 				}
 			}
-			//TODO: Add useLives back.
-			if (strValue[0] != '\0') //Value has something in it, by time we get here we already have all the value parsed(hopefully).
+			case 'n', 'x':
 			{
-				new Float:value = StringToFloat(strValue);
-				Debug("Operating on value");
-				switch(_operator[bracket])
-				{
-					case Operation_Add:
-					{
-						Debug("Adding %f onto %f", value, sum[bracket]);
-						sum[bracket] += value;
-					}
-					case Operation_Subtract:
-					{
-						Debug("Subtracting %f onto %f", value, sum[bracket]);
-						sum[bracket] -= value;
-					}
-					case Operation_Multiplication:
-					{
-						Debug("Multiplying %f onto %f", value, sum[bracket]);
-						sum[bracket] *= value;
-					}
-					case Operation_Division:
-					{
-						Debug("Running 0 check b4 dividing.");
-						int dotInd = StrContains(strValue, ".");
-						Debug("dotInd = %i", dotInd);
-						if (dotInd > -1)
-						{
-							Debug("'.' detected!");
-							int mult = strlen(strValue) - dotInd + 1;
-							Debug("strlen(strValue) = %i; mult = %i; inEnd = %f", strlen(strValue), mult, StringToFloat(strValue) * float(mult));
-							if (!RoundFloat(StringToFloat(strValue) * float(mult)))
-							{
-								bracket = 0;
-								sum[bracket] = -999999.0; //Let malformation handle this.
-								_operator[bracket] = Operation_None;
-								break;
-							}
-						}
-						else if (!StringToInt(strValue))
-						{
-							Debug("No '.' detected, got 0");
-							bracket = 0;
-							sum[bracket] = -999999.0; //Let malformation handle this.
-							_operator[bracket] = Operation_None;
-							break;
-						}
-						Debug("Dividing %f onto %f", value, sum[bracket]);
-						sum[bracket] /= value;
-					}
-					case Operation_Exponential:
-					{
-						Debug("Powering %f onto %f", value, sum[bracket]);
-						sum[bracket] = Pow(sum[bracket], value);
-					}
-					default: //Just set sum to be value. (eg: "(5000)" or incase of contant value.)
-					{
-						Debug("Setting %f onto %f", value, sum[bracket]);
-						sum[bracket] = value;
-					}
-				}
+				Operate(sum, bracket, float(playing), _operator);
 			}
-			strcopy(strValue, sizeof(strValue), "");
-			//_operator[bracket] = Operation_None;
 		}
 	}
-	new result=RoundFloat(sum[0]);
-	Debug("sum[0] = %f, result = %i", sum[0], result);
-	if(!result && strValue[0]!='\0') //Check to see if we're dealing with a constant health value
-	{
-		Debug("Constant detected after for...");
-		return StringToInt(strValue);
-	}
-	if(bracket || result<=0)
-	{
-		Debug("MALFORMATION! bracket=%i", bracket);
-		decl String:bossName[32];
-		KvRewind(BossKV[Special[client]]);
-		KvGetString(BossKV[Special[client]], "name", bossName, sizeof(bossName));
-		LogError("[FF2] %s has a malformed %s formula, using default!", bossName, key);
-		return defaultValue;
-	}
-	if(bMedieval)
-	{
-		return RoundFloat(result/3.6); //TODO: Make this configurable
-	}
-	return result;
+	float tSum = view_as<float>sum.Get(0);
```

Comment:
That was b4 I knew sm was fine with any to float
On 2015-03-21 9:20 PM, "Wliu" <notifications@github.com> wrote:

> In addons/sourcemod/scripting/freak_fortress_2.sp
> <https://github.com/50DKP/FF2-Official/pull/232#discussion_r26898795>:
>
> > -	}
> > -	if(bracket || result<=0)
> > -	{
> > -		Debug("MALFORMATION! bracket=%i", bracket);
> > -		decl String:bossName[32];
> > -		KvRewind(BossKV[Special[client]]);
> > -		KvGetString(BossKV[Special[client]], "name", bossName, sizeof(bossName));
> > -		LogError("[FF2] %s has a malformed %s formula, using default!", bossName, key);
> > -		return defaultValue;
> > -	}
> > -	if(bMedieval)
> > -	{
> > -		return RoundFloat(result/3.6); //TODO: Make this configurable
> > -	}
> > -	return result;
> > +	float tSum = view_as<float>sum.Get(0);
>
> ??
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/50DKP/FF2-Official/pull/232/files#r26898795>.
>